packages/shared/src/UI/components/TokenIcon/index.tsx

```diff
@@ -0,0 +1,92 @@
+import { makeStyles, Avatar, Theme, AvatarProps } from '@material-ui/core'
+import { useImageFailOver } from '../../index'
+import SPECIAL_ICON_LIST from './TokenIconSpecialIconList.json'
+import { useStylesExtends } from '../../UIHelper/custom-ui-helper'
+import {
+    ChainId,
+    currySameAddress,
+    formatEthereumAddress,
+    getTokenAssetBaseURLConstants,
+    getTokenConstants,
+    isSameAddress,
+    useBlockie,
+    useChainDetailed,
+} from '@masknet/web3-shared'
+import { memo } from 'react'
+
+//#region fix icon image
+function resolveTokenIconURLs(address: string, baseURIs: string[], chainId: ChainId, logoURI?: string) {
+    const checkSummedAddress = formatEthereumAddress(address)
+
+    if (isSameAddress(getTokenConstants().NATIVE_TOKEN_ADDRESS, checkSummedAddress)) {
+        return baseURIs.map((x) => `${x}/info/logo.png`)
+    }
+
+    const specialIcon = SPECIAL_ICON_LIST.find(currySameAddress(address))
+
+    if (specialIcon) return [specialIcon.logo_url]
+
+    const fullIconAssetURIs = baseURIs.map((x) => `${x}/assets/${checkSummedAddress}/logo.png`)
```

Comment:
Wait, are they local assets or remote assets? Please document it in the code.

Reply:
This is remote assets, I will leave comment in the code.